File: IDTECH/100_days_of_code/day2_check_palindrome.py
''' python function that check if a string
    is a palindrome or not'''

def checkPalindrome(string: str) -> bool:
    ''' This function takes a string parameter and check if palindrome or not.'''
    check = True
    newString = "".join(c for c in string if c.isalpha())
    newString = newString.lower()
    stringSecond = newString[::-1]
    #print(newString + " reversed string: " + stringSecond)
    if (newString == stringSecond):
        print(string + " is a palindrome")
    else: print(string + " is not a palindrome")
    return check

checkPalindrome("A man, a plan, a canal,Panama.")
checkPalindrome("Step on no pets.")
checkPalindrome("Was it a car or a cat I saw?")
checkPalindrome("Mr.Owl ate my metal worm.")
checkPalindrome("Never odd or even.")

File: IDTECH/100_days_of_code/day2_check_palindrome.js
/*
This is a javascriptfunction that check if a palindrome or not 
*/

function checkPalindrome(strings) {
  if (typeof(strings) != String) {
    console.log(typeof(strings) + " is not a string");
    return
  } else if (strings === "") {
    console.log(strings + " is an empty string");
    return
  } else {
    let newString = strings.replace(/[^a-z0-9]/gi, '');
  newString = newString.toLowerCase();
  reversedString = newString.split("").reverse().join("");
  if (newString == reversedString) {
    console.log("The sentence or word ==> " + strings + " is a palindrome");
  } else {
    console.log("The sentence or word ==> " + strings + " is not a palindrome")
  }
}
}
checkPalindrome("111");
checkPalindrome("");
checkPalindrome("check");
checkPalindrome("Mr.Owl ate my metal worm.");
checkPalindrome("Never odd or even.");
